tests/acceptance/stepDefinitions/loginContext.js

```diff
@@ -10,12 +10,18 @@ Given(/^the user has browsed to the login page$/,
       .navigate()
   })
 
+Given('the user has clicked the authenticate button', () => client.page.loginPage().authenticate())
+
 When('the user clicks the authenticate button', () => client.page.loginPage().authenticate())
 
 When('the user logs in with username {string} and password {string} using the webUI',
   (username, password) => client.page.ownCloudLoginPage().login(username, password)
 )
 
+When('the user tries to log in with username {string} and password {string} using the webUI',
+  (username, password) => client.page.ownCloudLoginPage().login(username, password)
```

Comment:
As it seems `ownCloudLoginPage().login()`  just clicks the login button after filling in the form, Shouldn't we also wait for some ajax calls or something. This way when we later check that the user is redirected to login page then we can verify that the user has been redirected there after login request was actually sent.